src/NavList/NavList.tsx

```diff
@@ -1,23 +1,27 @@
 import {ChevronDownIcon} from '@primer/octicons-react'
 import {useSSRSafeId} from '@react-aria/ssr'
 import React, {isValidElement} from 'react'
+import styled from 'styled-components'
 import {ActionList} from '../ActionList'
 import Box from '../Box'
 import StyledOcticon from '../StyledOcticon'
+import sx, {merge, SxProp} from '../sx'
 
 // ----------------------------------------------------------------------------
 // NavList
 
 export type NavListProps = {
   children: React.ReactNode
-} & React.ComponentProps<'nav'>
+} & SxProp &
+  React.ComponentProps<'nav'>
+
+const NavBox = styled.nav<SxProp>(sx)
```

Comment:
Does this go against ADR 005? https://github.com/primer/react/blob/main/contributor-docs/adrs/adr-005-box-sx.md

cc @siddharthkp

Reply:
I copied this from `ActionList` ([code](https://github.com/primer/react/blob/main/src/ActionList/List.tsx#L30)) as a workaround for a ref type error I ran into.

```jsx
// TypeScript doesn't like this
<Box as="nav" ref={ref} sx={{...}}>

// But this works
const NavBox = styled.nav(sx)
<NavBox ref={ref} sx={{...}}>
```

If there's another way to workaround that type issue, I'm open to updating this!

Reply:
I've also struggled with this type issue, and I'm not sure how to fix it 😅 

Let's leave this overnight and see if any of our coworkers in other timezones know a solution.

Reply:
> I copied this from ActionList ([code](https://github.com/primer/react/blob/main/src/ActionList/List.tsx#L30)) as a workaround for a ref type error I ran into.

Oh yes, I remember this! I don't have a better solution for this either :( 

I guess we should update the ADR with a note for components that accept a ref?

Reply:
I'm going to move forward with this for now. A long-term solution might be to move away from the `as` prop entirely 😅